engines/watchmaker/3d/animation.cpp

```diff
@@ -0,0 +1,730 @@
+	dest->M[7] = matrix.M[5];
+	dest->M[3] = matrix.M[6];
+	dest->M[5] = matrix.M[7];
+	dest->M[4] = matrix.M[8];
+}
+
+Common::Array<t3dPLIGHT> t3dBODY::getPositionalLight(uint8 pos) {
+	Common::Array<t3dPLIGHT> result;
+	for (auto light : PosLightTable) {
+		if (light.Num == pos) {
+			result.push_back(light);
+		}
+	}
+	return result;
+}
+
+/* -----------------04/07/98 15.52-------------------
+ *                  GetLightPosition
+ * --------------------------------------------------*/
+uint8 GetLightPosition(t3dV3F *dest, uint8 pos) {
+	if (!pos) return 0;
+
+	auto pLights = t3dCurRoom->getPositionalLight(pos);
+	dest->y = CurFloorY;
+	for (auto light : pLights) {
+		if (light.Pos.x && light.Pos.z) {
+			dest->x = light.Pos.x;
+			dest->z = light.Pos.z;
+			return pos;
+		}
+	}
+	if (pos != 99)
+		warning("Can't find lpos %d in %s", pos, t3dCurRoom->name.c_str());
+	return 0;
+}
+
+/* -----------------04/07/98 15.52-------------------
+ *                  GetLightPosition
+ * --------------------------------------------------*/
+uint8 GetLightDirection(t3dV3F *dest, uint8 pos) {
+	if (!pos) return 0;
+
+	auto pLights = t3dCurRoom->getPositionalLight(pos);
+	dest->y = CurFloorY;
+	for (auto light : pLights) {
+		if (light.Dir.x && light.Dir.z) {
+			dest->x = light.Dir.x;
+			dest->z = light.Dir.x;
+			return pos;
+		}
+	}
+	if (pos != 99)
+		warning("Can't find ldir %d in %s", pos, t3dCurRoom->name.c_str());
+	return 0;
+}
+
+
+/* -----------------15/12/98 16.26-------------------
+ *                  t3dLoadAnimation
+ * --------------------------------------------------*/
+int8 t3dLoadAnimation(WGame &game, const char *s, t3dMESH *mesh, uint16 Flag) {
+	uint32 nf, nb, i, j, k, h, older, len, ScaleAnim, CurPreloadedAnim;
+	t3dLOADANIM *p;
+	t3dLOADBONE *bone;
+	t3dBONEANIM *db;
+	t3dBONE *b;
+	t3dV3F t;
+	t3dF32 c;
+
+//	Prova a vedere se l'ho gia' precaricata
+	for (CurPreloadedAnim = 0; CurPreloadedAnim < MAX_PRELOADED_ANIMS; CurPreloadedAnim++)
+		if (PreloadedAnim[CurPreloadedAnim].NumFrames)
+			if (PreloadedAnim[CurPreloadedAnim].name.equalsIgnoreCase(s))
+				break;
+//	Se la devo precaricare, cerco quella piu' vecchia e la scarico
+	if (CurPreloadedAnim >= MAX_PRELOADED_ANIMS) {
+		older = 0;
+//		Prima cerco se ci sono ancora degli slot liberi
+		for (CurPreloadedAnim = 0; CurPreloadedAnim < MAX_PRELOADED_ANIMS; CurPreloadedAnim++) {
+			if (!PreloadedAnim[CurPreloadedAnim].NumFrames)
+				break;
+			else if (!(older) || (older > PreloadedAnim[CurPreloadedAnim].LastTime))
+				older = PreloadedAnim[j = CurPreloadedAnim].LastTime;
+		}
+//		Se non c'erano slot liberi, rilascia vecchia animazione precaricata
+		if (CurPreloadedAnim >= MAX_PRELOADED_ANIMS) {
+			CurPreloadedAnim = j;
+//t         DebugFile( "Precarico animazione %s nello slot %d occupato da %s", s, CurPreloadedAnim, PreloadedAnim[j].Name );
+//			Disalloca tutto
+			for (i = 0; i < MAX_BONES; i++) {
+				t3dFree(PreloadedAnim[j].Bone[i].Trasl);
```

Comment:
Mass renaming is required for this struct members as well, e.g. Trasil, Euler, others. Could be done after the merge.